### ISSUE TITLE
`auto` stops at first `package.json` even without `engines.node`

### ISSUE THREAD

**Issue by mhassan1** [tj/n#636]
# Bug Report

## Summary

`n exec auto` stops at the first `package.json` it sees, even if that `package.json` does not contain `engines.node`

## Steps to Reproduce

1. Create directories `root` and `root/pkg`
2. Create a file `root/.nvmrc` containing `14`
3. Create a file `root/pkg/package.json` containing `{}`
4. In `root/pkg`, run `n exec auto node -v`

## Expected Behaviour

```
       found : /path/to/root/.nvmrc
        read : 14
v14.15.1
```

## Actual Behaviour

```
       found : /path/to/root/pkg/package.json
        read :
v15.3.0
```

## Configuration Details

```bash
$ n --version
6.7.0

$ command -v node
/Users/me/.n/bin/node

$ node -p process.platform
darwin
```

**Comment by shadowspawn** [comment]
(Nice description, thanks.)

That is the intended behaviour. `n` is looking for a file to consult, and stops searching at the first file it finds. This is similar to the behaviour of `npm` itself looking for the root of the package from the current directory.

- https://docs.npmjs.com/cli/v6/configuring-npm/folders#more-information

**Comment by mhassan1** [comment]
I have a monorepo with a structure like this:
```
.nvmrc
package.json
packages/
  package1/
    package.json
  package2/
    package.json
```
None of the `package.json` files have `engines.node` specified in them.

If I run `n exec auto node index.js` from within `packages/package1`, I expect it to use the version specified in `.nvmrc`. Since I haven't specified `engines.node` in the `packages/package1/package.json`, I wouldn't expect that to mean "use latest". At least for my use case, this behavior makes `auto` less useful, since I'd like to control the node version at the root without updating `engines.node` in every `package.json` file.

I think my bias here is that I am looking to get similar behavior to what I get with `nvm`, where it looks for the nearest `.nvmrc` file. Perhaps there's a feature request here where we can tweak `auto` to do the same directory traversal but only look for some file types. If you think that idea has potential, I can open a separate issue.

**Comment by shadowspawn** [comment]
With that setup and bias, I see why you hoped it would work that way. 😄 

I did wonder about what to do with missing `node.engines` when I wrote this feature. Installing `current` is loosely consistent with absence of `node.engines` but perhaps not particularly useful in the absence of any constraint. I didn't want to throw an error.

Continuing the search is different than I had in mind but reasonable, and useful for mono-repo type layouts. I'll think about that for a bit...
